src/main/java/au/com/centrumsystems/hudson/plugin/buildpipeline/extension/BuildCardExtension.java

```diff
@@ -201,6 +203,52 @@ public int triggerManualBuild(ItemGroup pipelineContext, Integer upstreamBuildNu
         return triggerBuild(triggerProject, upstreamBuild, buildParametersAction);
     }
 
+    /**
+     * Begin a recursive search on ItemGroup
+     * @param groupToTraverse - the context that should be searched
+     * @param searchTerm - name of the item that is being looked for
+     * @return the project that matched the search term
+     */
+    private AbstractProject<?, ?> traverseItemGroup(ItemGroup groupToTraverse, String searchTerm)
+    {
+        final AbstractProject<?, ?> helperResult = traverseItemGroupHelper(groupToTraverse, searchTerm);
+        if (helperResult == null) {
+            throw new NoSuchElementException("Item with name " + searchTerm
+                    + " was not found in ItemGroup " + groupToTraverse.getFullDisplayName());
+
+        } else {
+            return helperResult;
+        }
+    }
+
+    /**
+     * Recursively search the ItemGroup tree
+     * @param groupToTraverse - the context that should be searched
+     * @param searchTerm - name of the item that is being looked for
+     * @return the project that matched the search term
+     */
+    private  AbstractProject<?, ?> traverseItemGroupHelper(ItemGroup groupToTraverse, String searchTerm)
```

Comment:
There is a boundary case here where there can be a job with the same name (say, "Prod Commit") but in separate folders. This will always return the first folder to contain a job named "Prod Commit".

However this is also not where the issue is - we are able to correctly identify the upstream project and upstream build from the information the JS passes us